CMakeLists.txt

```diff
@@ -36,6 +22,41 @@ if(COMMAND cmake_policy)
     endif()
 endif(COMMAND cmake_policy)
 
+
+##################################################################################################################
+# VERSION
+##################################################################################################################
+
+set(ALLEGRO_VERSION_REGEX_PATTERN "#define ALLEGRO_VERSION_STR[ ]+\"([0-9]+\\.[0-9]+\\.[0-9]+).*\"")
```

Comment:
> This seems kind of backwards, usually the build system configures the source files, not the other way around. What makes versioning different?

I think this topic may also be subjective, but something worth noting and maybe a bit nit-picking; cmake isn't a build system (maybe in the abstract idea), but rather a meta-build system. It generates build system files.
As for what would be the right approach, IMHO I think it makes sense modifying version in one place (assuming the fixver.sh script wasn't there for the rescue). And ideally in the source files rather than a meta-build system script. Generally because projects may have multiple build system files, and yes, I've seen this in some big projects that don't use a meta-build system generator, such as cmake. Then you would have to specify version in each script, instead of making the scripts parsing the version from a single header file as the source of truth.

So to sum-up:
Old behavior of versioning:
1. Modify CMakeLists.txt version
2. Run the script fixver.sh so that it updates the versions in base.h file

New behavior of versioning:
1. Run the script fixver.sh with new version

I personally find the new behavior easier to maintain and less prone to human error, but I might be biased ;)